src/riak_kv_vnode.erl

```diff
@@ -1686,15 +1702,44 @@ handle_aaefold({merge_branch_nval, Nval, BranchIDs},
                                         ReturnFun),
     {noreply, State};
 handle_aaefold({fetch_clocks_nval, Nval, SegmentIDs}, 
-                    _InitAcc, Nval,
-                    IndexNs, _Filtered, ReturnFun, Cntrl, _Sender,
+                    InitAcc, Nval,
+                    IndexNs, _Filtered, ReturnFun, Cntrl, Sender,
                     State) ->
-    aae_controller:aae_fetchclocks(Cntrl, 
-                                    IndexNs, 
-                                    SegmentIDs, 
-                                    ReturnFun, 
-                                    fun preflistfun/2),
-    {noreply, State};
+    case app_helper:get_env(riak_kv, aae_fetchclocks_repair, false) of
+        true ->
+            aae_controller:aae_fetchclocks(Cntrl, 
+                                            IndexNs, 
+                                            SegmentIDs, 
+                                            ReturnFun, 
+                                            fun preflistfun/2),
+            
+            {noreply, State};
+        false ->
+            %% Using fetch_clocks_range will mean that the AF3_QUEUE will be
+            %% used for scheduling the work not the aae_runner.  Also, the 
+            %% fetch clock query will not attempt to rebuild segments of the
+            %% tree cache - this is left to internal anti-entropy
+            handle_aaefold({fetch_clocks_range,
```

Comment:
kind of too many arguments to be able to really get what's going on here.

Reply:
you could equally well have called with fetch_clocks_nvals, since that directly returns in this new call.

Reply:
This way just helps with concurrently managing the update to 2.9 and 3.0, as 2.9 won't initially have the new fetch_clocks_nval option